### PR TITLE
Handle shutdown from jupyter_client

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -48,6 +48,11 @@ void handler(int sig)
 }
 #endif
 
+void stop_handler(int /*sig*/)
+{
+    exit(0);
```
Comment (martinRenou) [inline src/main.cpp:53]: @JohanMabille @SylvainCorlay should we do a proper `server.stop` in the handler?

Comment (JohanMabille) [reply src/main.cpp:53]: I think it's not possible, you are very limited in what you can do inside a signal handler.

Comment (martinRenou) [reply src/main.cpp:53]: Yeah that's what I was seing... 

I see that the xserver in xeus does not implement a destructor, should we implement a destructor that calls `stop`?

Comment (JohanMabille) [reply src/main.cpp:53]: No, you mustn't call a virtual method from a destructor (`stop calls `stop_impl` in the end). And the `stop_impl` method may throw, which would prevent the destruction of the sockets.

Comment (martinRenou) [reply src/main.cpp:53]: Ok. So I guess my question now is: is it fine to `exit(0)` without calling `server.stop()`?

Comment (JohanMabille) [reply src/main.cpp:53]: Yes.

**Pull request merged**